sentry/sentry.go

```diff
@@ -0,0 +1,84 @@
+package sentry
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/fabric8-services/fabric8-wit/token"
+
+	"github.com/getsentry/raven-go"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+)
+
+// Client encapsulates client to Sentry service
+// also has mutex which controls access to the client
+type Client struct {
+	c   *raven.Client
+	mux sync.Mutex
+}
+
+var (
+	sentryClient *Client
+)
+
+// Sentry returns client declared inside package
+func Sentry() *Client {
+	return sentryClient
+}
+
+// InitializeSentryClient initializes sentry client
+func InitializeSentryClient(release, environment string) error {
+	c, err := raven.New(os.Getenv("SENTRY_DSN"))
+	if err != nil {
+		return err
+	}
+
+	c.SetRelease(release)
+	c.SetEnvironment(environment)
+	sentryClient = &Client{
+		c: c,
+	}
+
+	return nil
+}
+
+// CaptureError sends error 'err' to Sentry, meanwhile also sets user
+// information by extracting user information from the context provided
+func (c *Client) CaptureError(ctx context.Context, err error) error {
+	// extract user information
+	user, errLocal := extractUserInfo(ctx)
```

Comment:
is it okay if I just log it here: https://github.com/fabric8-services/fabric8-wit/pull/1909/files#diff-b160c561ec25cc415f1cdbe6d0bebb8dR172 ?

Since we need to handle it somewhere?

Reply:
Why not just clean up the user context and still submit to sentry? Otherwise you won't submit anything to sentry for 500 happened in endpoints which do not require any tokens.

Reply:
done as you recommended!